pkg/types/azure/machinepool.go

```diff
@@ -15,6 +15,12 @@ type MachinePool struct {
 	// +optional
 	InstanceType string `json:"type"`
 
+	// HyperVGeneration is the field that determines the version of HyperVGeneration of the disk.
+	// eg. V1, V2
+	//
+	// +optional
+	HyperVGeneration string `json:"hypervgeneration,omitempty"`
```

Comment:
Do we need to let users specify this? if we default to gen2 when gen2 is available and gen1 for gen1 only instances, the only case this is needed is if users want to use gen1 with a gen2 capable instance. From my understanding of the generations, wanting to downgrade is unlikely. There could be an escape hatch through editing the machine manifests in the unlikely event a downgrade is desired.

Reply:
I'll try to bypass the install config.